my/core/__main__.py

```diff
@@ -360,69 +329,121 @@ def module_install(args: Namespace) -> None:
     check_call(cmd)
 
 
-from argparse import ArgumentParser
-def parser() -> ArgumentParser:
-    p = ArgumentParser('Human Programming Interface', epilog='''
-Tool for HPI.
+@click.group()
+def main() -> None:
+    '''
+    Human Programming Interface
 
-Work in progress, will be used for config management, troubleshooting & introspection
-''')
-    sp = p.add_subparsers(dest='mode')
-    dp = sp.add_parser('doctor', help='Run various checks')
-    dp.add_argument('--verbose', action='store_true', help='Print more diagnosic infomration')
-    dp.add_argument('--all'    , action='store_true', help='List all modules, including disabled')
-    dp.add_argument('--quick'  , action='store_true', help='Only run partial checks (first 100 items)')
-    dp.add_argument('module', nargs='?', type=str   , help='Pass to check a specific module')
-    dp.set_defaults(func=doctor)
+    Tool for HPI
+    Work in progress, will be used for config management, troubleshooting & introspection
+    '''
+    # note: code in groups doesnt run unless a subcommand is passed
+    # https://click.palletsprojects.com/en/7.x/commands/#group-invocation-without-command
+    # https://click.palletsprojects.com/en/7.x/commands/#multi-command-chaining
 
-    cp = sp.add_parser('config', help='Work with configuration')
-    scp = cp.add_subparsers(dest='config_mode')
-    if True:
-        ccp = scp.add_parser('check', help='Check config')
-        ccp.set_defaults(func=config_check_cli)
+    # use a particular directory instead of a random one, since
+    # click being decorator based means its more complicated
+    # to run things at the end (would need to use a callback or pass context)
+    # https://click.palletsprojects.com/en/7.x/commands/#nested-handling-and-contexts
 
-        icp = scp.add_parser('create', help='Create user config')
-        icp.set_defaults(func=config_create)
+    import tempfile
 
-    mp = sp.add_parser('modules', help='List available modules')
-    mp.add_argument('--all'    , action='store_true', help='List all modules, including disabled')
-    mp.set_defaults(func=list_modules)
+    tdir: str = os.path.join(tempfile.gettempdir(), 'hpi_temp_dir')
+    if not os.path.exists(tdir):
+        os.makedirs(tdir)
+    os.chdir(tdir)
 
-    op = sp.add_parser('module', help='Module management')
-    ops = op.add_subparsers(dest='module_mode')
-    if True:
-        add_module_arg = lambda x: x.add_argument('module', type=str, help='Module name (e.g. my.reddit)')
 
-        opsr = ops.add_parser('requires', help='Print module requirements')
-        # todo not sure, might be worth exposing outside...
-        add_module_arg(opsr)
-        opsr.set_defaults(func=module_requires)
+@main.command(name='doctor', short_help='run various checks')
+@click.option('--verbose/--quiet', default=False, help='Print more diagnostic information')
```

Comment:
>  options with '/' are interpreted as on/off flags

uhh neat -- didn't know

Reply:
Often described in the docs like `--save/--no-save` in the docs; is why that doesnt have the `is_flag=True` option